plugin/pkg/scheduler/factory/factory.go

```diff
@@ -168,14 +169,41 @@ func (factory *ConfigFactory) createMinionLW() *cache.ListWatch {
 	}
 }
 
-// pollMinions lists all minions and returns an enumerator for cache.Poller.
+// pollMinions lists all minions and filter out unhealthy ones, then returns
+// an enumerator for cache.Poller.
 func (factory *ConfigFactory) pollMinions() (cache.Enumerator, error) {
-	list := &api.NodeList{}
-	err := factory.Client.Get().Resource("minions").Do().Into(list)
+	allNodes := &api.NodeList{}
+	err := factory.Client.Get().Resource("minions").Do().Into(allNodes)
 	if err != nil {
 		return nil, err
 	}
-	return &nodeEnumerator{list}, nil
+	nodes := &api.NodeList{
+		TypeMeta: allNodes.TypeMeta,
+		ListMeta: allNodes.ListMeta,
+	}
+	for _, node := range allNodes.Items {
+		if len(node.Status.Conditions) == 0 {
+			// If no condition is set, node health check is disabled (master flag
+			// healthCheckMinions is set to false), we add nodes unconditionally.
+			glog.Infof("No condition set for node %v", node.Name)
+			nodes.Items = append(nodes.Items, node)
+			continue
+		}
+		for _, condition := range node.Status.Conditions {
+			// TODO: Distinguish different conditions and implement better condition check.
+			switch condition.Kind {
+			case api.NodeReady:
+				if condition.Status == api.ConditionFull {
+					nodes.Items = append(nodes.Items, node)
+				} else {
+					glog.V(2).Infof("Ignore unhealthy node %v", node.Name)
+				}
+			default:
+				glog.Errorf("Unknown condition kind %v for node %v", condition.Kind, node.Name)
```

Comment:
The reason I'm logging unknown type of status is that *all* conditions are well defined and will only be updated by core component.  There shouldn't be any unknown condintions if the system operates correctly.  I'm ok to not log it as scheduler doesn't care about such condition, but would like to know why you don't like it :)

Will fix the code shortly. Going for dinner.